lib/msal-node/src/client/ManagedIdentityClient.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    CacheManager,
+    DEFAULT_CRYPTO_IMPLEMENTATION,
+    DefaultStorageClass,
+    INetworkModule,
+    Logger,
+    LoggerOptions,
+    NetworkManager,
+} from "@azure/msal-common";
+import { ManagedIdentityIdType } from "../utils/Constants";
+import { ServiceFabric } from "./ManagedIdentitySources/ServiceFabric";
+import { AppService } from "./ManagedIdentitySources/AppService";
+import { CloudShell } from "./ManagedIdentitySources/CloudShell";
+import { AzureArc } from "./ManagedIdentitySources/AzureArc";
+import { Imds } from "./ManagedIdentitySources/Imds";
+import {
+    ManagedIdentityConfiguration,
+    ManagedIdentityNodeConfiguration,
+    buildManagedIdentityConfiguration,
+} from "../config/Configuration";
+import { version, name } from "../packageMetadata.js";
+
+/**
+ * Class to initialize a managed identity and identify the service
+ */
+export class ManagedIdentityClient {
```

Comment:
Will this work though? For example are you going to be able to reuse the token caching logic that is already tied to ConfidentialClientApplication?